### PR TITLE
[FIX] mail,sms: restore "onchange_on_keydown" feature

### PR DESCRIPTION
**Pull request by awa-odoo**
Previously, we had an option on char/text fields called "onchange_on_keydown", which allowed to trigger the onchange while the user was typing (instead of when the field is blurred).

It was notably used when typing a phone number in the sms composer, to warn the user that the phone number is invalid.

This feature was lost during the port to OWL, this commit restores it.

Task-3033107

### CODE REVIEW

File: addons/mail/static/src/js/onchange_on_keydown.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/** @odoo-module **/
+
+import { patch } from "@web/core/utils/patch";
```
Comment (std-odoo) [inline addons/mail/static/src/js/onchange_on_keydown.js:3]: can we add a small test to check that the onchange value is correct ? (e.g. if the input is empty and you press "e", the event.value should be "e", not the "previous value" of the input)

File: addons/mail/static/src/js/onchange_on_keydown.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * (or 2 seconds by default) when typing ends.
+ *
+ */
+const onchangeOnKeydownMixin = {
+    setup() {
+        this._super(...arguments);
+
+        if (this.props.onchangeOnKeydown) {
+            const input = this.input || this.textareaRef;
+            this.triggerOnChange = debounce(this.triggerOnChange, this.props.keydownDebounceDelay);
```
Comment (std-odoo) [inline addons/mail/static/src/js/onchange_on_keydown.js:23]: `triggerOnChange` is used only when the option is true ?

If yes, I would not modify `this.triggerOnChange` and make a variable instead

```js
const triggerOnChange = debounce(this.triggerOnChange, this.props.keydownDebounceDelay);
```

File: addons/sms/wizard/sms_composer_views.xml [hunk 3]
```diff
@@ -43,7 +43,7 @@
                             attrs="{'invisible': [('comment_single_recipient', '=', False)]}"/>
                         <div attrs="{'invisible': [('comment_single_recipient', '=', False)]}">
                             <field name="recipient_single_description" class="oe_inline"/>
-                            <field name="recipient_single_number_itf" class="oe_inline" nolabel="1" options="{'onchange_on_keydown': True}" placeholder="e.g. +1 415 555 0100"/>
+                            <field name="recipient_single_number_itf" class="oe_inline" nolabel="1" onchange_on_keydown="true" placeholder="e.g. +1 415 555 0100"/>
```
Comment (std-odoo) [inline addons/sms/wizard/sms_composer_views.xml:46]: picky std, because it's a python view I would use `True` (even it does exactly the same)

**Review by std-odoo — COMMENTED** [review]
@awa-odoo Hi :) 

I don't have that much to say, except it might deserve a small test, otherwise LGTM :)

**Review by awa-odoo — COMMENTED** [review]
Hello @std-odoo 

I addressed your comments and added a QUnit test.
Not sure it's what you had in mind but it should cover the use case pretty well.

Thanks & cheers!

**Review by std-odoo — APPROVED** [review]
@awa-odoo Yes that's what I had in mind, thanks for the update, LGTM :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+

Thanks @awa-odoo and @std-odoo

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**